tools/kafka_ssl/generate_ssl_CA/ssl_renewal.sh

```diff
@@ -0,0 +1,269 @@
+            ;;
+        --gen_CARoot)
+            CARoot_flag=true
+            ;;
+        --gen_subCA)
+            subCA_flag=true
+            ;;
+        --gen_keystore)
+            genKeystore_flag=true
+            ;;
+        --gen_truststore)
+            genTruststore_flag=true
+            ;;
+        --import_to_keystore)
+            importToKeystore_flag=true
+            ;;
+        --import_to_truststore)
+            importToTruststore_flag=true
+            ;;
+        --cer)
+            CARoot=`path_conver $2`
+            shift
+            ;;
+        --cerKey)
+            CA_key=`path_conver $2`
+            shift
+            ;;
+        --keystore)
+            keystore=`path_conver $2`
+            shift
+            ;;
+        --truststore)
+            truststore=`path_conver $2`
+            shift
+            ;;
+        --import)
+            importCA=`path_conver $2`
+            shift
+            ;;
+        -d|--directory)
+            generate_root=$2
+            if [ ! -d ${generate_root} ]; then
+              error "The path '$generate_root' does not exist"
+              exit 1
+            else
+              generate_root=`path_conver $generate_root`/SSL_files
+            fi
+            shift
+            ;;
+        -p|--passphrase)
+            pass=$2
+            if [ ${#pass} -lt 6 ];then
+              error "Password is too short - must be at least 6 characters."
+              exit 1
+            fi
+            shift
+            ;;
+        -c|--CN)
+            CN=$2
+            shift
+            ;;
+        -s|--storetype)
+            storetype=$2
+            shift
+            ;;
+        -i|--install)
+            SETUP_JDK=true
+            SETUP_OPENSSL=true
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            error "${bldred}Invalid option, the correct usage is described below: $txtrst"
+            general_help
+            ;;
+    esac
+shift
+done
+
+if [[ ! -z $help_flag ]]; then
+  if [[ ! -z $CARoot_flag ]]; then
+    general_usage gen_CARoot
+  elif [[ ! -z $subCA_flag ]]; then
+    general_usage gen_subCA
+  elif [[ ! -z $genKeystore_flag ]]; then
+    general_usage gen_keystore
+  elif [[ ! -z $genTruststore_flag ]]; then
+    general_usage gen_truststore
+  elif [[ ! -z $importToKeystore_flag ]]; then
+    general_usage import_to_keystore
+  elif [[ ! -z $importToTruststore_flag ]]; then
+    general_usage import_to_truststore
+  else
+    general_help
+  fi
+  exit 0
+else
+  # this script only support rhel/centos
+  prog "Checking operation system (OS) version ..."
+  check_os $OS $version
+
+  prog "Checking root/sudo ..."
+  check_root
+
+  # Using option '-i/--install' will install openjdk-1.8.0 and openssl,
+  # otherwise openjdk-1.8.0 and openssl will not be installed
+  # install openJDK
+  install_openJDK
+  # install openssl
+  install_openssl
+
+  # generate root CA
+  if [[ ! -z $CARoot_flag ]]; then
```

Comment:
why do they need to generate root CA for renewal case? you said this is for who has rootCA or subCA already?